### ISSUE TITLE
Incorrect schema in redirect behind reverse proxy

### ISSUE THREAD

**Issue by m0nhawk** [OHDSI/WebAPI#1901]
### Expected behavior

All the redirects should have HTTPS schema.

### Actual behavior

Pac4j redirects to `http://internal.domain/WebAPI/user/login/openid?redirectUrl=/search` in OIDC flow.

### Steps to reproduce behavior

For test purposes I'm trying to run the Atlas and WebAPI in Kubernetes, behind nginx as a reverse proxy, with the following config:

```
security_cors_enabled: "true"
security_origin: "*"
security_token_expiration: "43200"
security_ssl_enabled: "false"

security_provider: AtlasRegularSecurity

security_oid_clientId: [omitted]
security_oid_apiSecret: [omitted]
security_oid_url: https://login.internal.domain/.well-known/openid-configuration
security_oid_redirectUrl: https://internal.domain/atlas/#/welcome
security_oid_logoutUrl: https://internal.domain/atlas/#/home

security_oauth_callback_ui: https://internal.domain/atlas/#/welcome
security_oauth_callback_api: https://internal.domain/WebAPI/user/oauth/callback
security_oauth_callback_urlResolver: query
```

All the callbacks/URL's are set to HTTPS, but in the login flow it's redirecting to HTTP, with the following logs.

```
2021-06-19 03:56:18.445 DEBUG http-nio-8080-exec-1 org.pac4j.core.engine.savedrequest.DefaultSavedRequestHandler -  - requestedAction: Performing a 302 HTTP action
2021-06-19 03:56:18.560 DEBUG http-nio-8080-exec-3 org.pac4j.core.engine.DefaultSecurityLogic -  - === SECURITY ===
2021-06-19 03:56:18.560 DEBUG http-nio-8080-exec-3 org.pac4j.core.engine.DefaultSecurityLogic -  - url: http://internal.domain/WebAPI/user/login/openid?redirectUrl=/search
2021-06-19 03:56:18.560 DEBUG http-nio-8080-exec-3 org.pac4j.core.engine.DefaultSecurityLogic -  - matchers: null
2021-06-19 03:56:18.560 DEBUG http-nio-8080-exec-3 org.pac4j.core.engine.DefaultSecurityLogic -  - clients: OidcClient
2021-06-19 03:56:18.560 DEBUG http-nio-8080-exec-3 org.pac4j.core.client.finder.DefaultSecurityClientFinder -  - Provided clientNames: OidcClient
2021-06-19 03:56:18.561 DEBUG http-nio-8080-exec-3 org.pac4j.core.client.finder.DefaultSecurityClientFinder -  - clientNameOnRequest: Optional.empty
2021-06-19 03:56:18.561 DEBUG http-nio-8080-exec-3 org.pac4j.core.client.finder.DefaultSecurityClientFinder -  - result: [OidcClient]
```

---

I'm not sure why it uses HTTP as a schema and can't find anything relevant in WebAPI wiki or Pac4j docs (or from where it gets the schema [here](https://github.com/pac4j/pac4j/blob/ae82c3ece16963a27b0955ef8c3b51a90fc23ee3/pac4j-core/src/main/java/org/pac4j/core/engine/DefaultSecurityLogic.java#L83-L86)).

**Comment by chgl** [comment]
I've been playing around with OpenID (via Keycloak) and the WebAPI for some time as well, I've documented what finally worked as part of the Helm chart I'm using: https://github.com/chgl/charts/tree/master/charts/ohdsi#securing-atlas-using-openid-connect.

It might be:

```yaml
- name: SECURITY_OID_REDIRECTURL
  value: "https://omop.example.com/atlas/index.html#/welcome/null"
```

note the `/null` which I picked up from some forum entry or GitHub issue.

Are you using the nginx Ingress controller?
